File: JavaFundamentals/src/MapsLambdaAndStreamAPI/Lab/WordSynonyms.java
package MapsLambdaAndStreamAPI.Lab;

import java.util.*;

public class WordSynonyms {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int count = Integer.parseInt(scanner.nextLine());

        Map<String, ArrayList<String>> synonyms = new LinkedHashMap<>(count);

        for (int i = 0; i < count; i++) {
            String word = scanner.nextLine();
            String synonym = scanner.nextLine();

            synonyms.computeIfAbsent(word, k -> new ArrayList<>()).add(synonym);
        }

        StringBuilder result = new StringBuilder();
        for (var entry : synonyms.entrySet()) {
            result.append(entry.getKey())
                    .append(" - ")
                    .append(String.join(", ", entry.getValue()))
                    .append("\n");
        }

        System.out.print(result);
    }
}

File: Fundamentals/Dictionaries - Lab/word_synonyms.py
count = int(input())
synonyms_words = dict()
for i in range(count):
    word = input()
    synonyms = input()
    if word not in synonyms_words.keys():
        synonyms_words[word] = list()
    synonyms_words[word].append(synonyms)

for word in synonyms_words:
    print(f"{word} - {', '.join(synonyms_words[word])}")